### ISSUE TITLE
[BUG] - Can not build Cardano on GHC latest

### ISSUE THREAD

**Issue by ssoto2** [input-output-hk/cardano-node#2351]
**Internal/External**
*Internal*   if an IOHK staff member.
*External*   otherwise.

**Area**
*Token Locking*    Related to Token Locking (Allegra).
*Native Tokens*    Related to Native Tokens (Mary).
*Other*   Any other topic (Delegation, Ranking, ...).

**Summary**
when attempting to build on with GHC 9.0.1 (Current Latest) Build Errors out 

**Steps to reproduce**
1. build ghc-9.0.1-x86_64-centos7-linux.tar.xz from https://downloads.haskell.org/ghc/latest/
2. run through the install and till this point 'cabal configure --with-compiler=/usr/local/bin/ghc-9.0.1'
3. observe error

**Expected behavior**
It would be expected that the system would configure

**System info (please complete the following information):**
- OS: Amazon Linux
- Version Amazon Linux 2
- Node 1.25.1

**Screenshots and attachments**
`Warning: Unknown/unsupported 'ghc' version detected (Cabal 3.2.0.0 supports
'ghc' version < 8.12): /usr/local/bin/ghc-9.0.1 is version 9.0.1
Warning: Requested index-state2021-01-10T00:00:00Z is newer than
'hackage.haskell.org'! Falling back to older state (2021-01-09T22:55:53Z).
Resolving dependencies...
cabal: Could not resolve dependencies:
[__0] trying: Win32-network-0.1.0.0 (user goal)
[__1] next goal: base (dependency of Win32-network)
[__1] rejecting: base-4.15.0.0/installed-4.15.0.0 (conflict: Win32-network =>
base>=4.5 && <4.15)
[__1] rejecting: base-4.14.1.0, base-4.14.0.0, base-4.13.0.0, base-4.12.0.0,
base-4.11.1.0, base-4.11.0.0, base-4.10.1.0, base-4.10.0.0, base-4.9.1.0,
base-4.9.0.0, base-4.8.2.0, base-4.8.1.0, base-4.8.0.0, base-4.7.0.2,
base-4.7.0.1, base-4.7.0.0, base-4.6.0.1, base-4.6.0.0, base-4.5.1.0,
base-4.5.0.0, base-4.4.1.0, base-4.4.0.0, base-4.3.1.0, base-4.3.0.0,
base-4.2.0.2, base-4.2.0.1, base-4.2.0.0, base-4.1.0.0, base-4.0.0.0,
base-3.0.3.2, base-3.0.3.1 (constraint from non-upgradeable package requires
installed instance)
[__1] fail (backjumping, conflict set: Win32-network, base)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: base, Win32-network
`

**Additional context**
# OS: AWSv2 
# CABAL: 3.2.0.0
# GHC: 9.0.1
# LIBSODIUM 66f017f1
# CARDANO-NODE: 1.25.1

**Comment by erikd** [comment]
Unfortunately `ghc-9.0` is not yet supported. The recommended GHC versions to build the node are `8.6.5` and `8.10.3/4`.

We will definitely not start supporting `ghc-9.0` until after we drop support for `ghc-8.6`.

**Comment by ssoto2** [comment]
@erikd that is good to know thanks.  Is there anywhere that says this because https://docs.cardano.org/projects/cardano-node/en/latest/getting-started/install.html# references 8.10.2 but also says that was the latest at the time of writing, but does not say it must be the used version or what is currently acceptable.

**Comment by jbarros35** [comment]
for some reason the command cabal configure --with-compiler=ghc-8.10.2 
takes forever to return, I mean it does not return anymore. I'm following the same guide

**Comment by rere-dnaw** [comment]
@jbarros35 try with:  cabal configure --with-compiler=ghc-8.6.5

**Comment by jbarros35** [comment]
solved, for anyone who had the same issue, I reinstalled the GHC and stick with the correct version 8.6.5. The rest of process is straightforward.

**Comment by irvlatta** [comment]
I'm still having this same problem.  I have tried with GSH 8.6.5 and 8.10.2 and 8.10.4.  They all run into the same problem when trying to run cabal configure --with-compiler=ghc-x.x.x.  The update begins but then gets stuck while trying to resolve dependencies and never completes.  The error message is Warning: Requested index-state2021-01-10T00:00:00Z is newer than
'hackage.haskell.org'! Falling back to older state (2021-01-09T22:55:53Z).
Resolving dependencies...

I'm following the instructions and have tried several times, even terminating the AWS instance and starting fresh multiple times.  Similar to above, I'm running AWSv2, cabal 3.2.0.0, libsodium 66f017f1, Cardano node 1.25.1.  I also tried the same steps with Cardano node 1.24.2 and it still gets hung up in the exact same place.

**Comment by libert-xyz** [comment]
Same issue in Ubuntu 20.04 following the instructions: https://docs.cardano.org/projects/cardano-node/en/latest/getting-started/install.html

**Comment by Frank-Buss** [comment]
I have the same error. I followed the official installation guide, and I get this at the end on my Debian system:
```
frank@hal9000:~/cardano-src/cardano-node$ git checkout tags/1.27.0
Note: checking out 'tags/1.27.0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 8fe46140a Merge #2671
frank@hal9000:~/cardano-src/cardano-node$ cabal configure --with-compiler=ghc-8.10.4
Warning: Unknown/unsupported 'ghc' version detected (Cabal 2.2.0.1 supports
'ghc' version < 8.5): /home/frank/.ghcup/bin/ghc-8.10.4 is version 8.10.4
Warning: Parsing the index cache failed (01-index.cache: unexpected magic
marker encountered: 11914548216615824884). Trying to regenerate the index
cache...
cabal: No cabal file found.
Please create a package description file <pkgname>.cabal
frank@hal9000:~/cardano-src/cardano-node$ echo "package cardano-crypto-praos" >>  cabal.project.local
frank@hal9000:~/cardano-src/cardano-node$ echo "  flags: -external-libsodium-vrf" >>  cabal.project.local
frank@hal9000:~/cardano-src/cardano-node$ cabal build all
cabal: No cabal file found.
Please create a package description file <pkgname>.cabal
```

**Comment by Frank-Buss** [comment]
Problem was that I had a system global cabal installed as well, with an older version, and when running ghcup, I chose to append it to the path. When prepending it to the path (or deleting the system global cabal), then I got this error:
https://github.com/input-output-hk/cardano-node/issues/1251
Changing it to ghc 8.6.5 instead of 8.10.4 didn't work for me. I think I'll try it again on a clean machine or Docker container.
